### ISSUE TITLE
Add --list=NUMBER support

### ISSUE THREAD

**Issue by glensc** [bulletmark/cdhist#8]
I would like to display not more than my terminal height (minus prompt):

```
$ cdhist --list=$((LINES-2))
usage: cdhist [-i] [-h] [-p] [-a] [-g] [-r] [-u] [-l] [-m SIZE] [-L] [-P] [directory]
cdhist: error: argument -l/--list: ignored explicit argument '31'
```

The idea is to limit output of cdhist by a NUMBER.

**Comment by bulletmark** [comment]
Why?

**Comment by glensc** [comment]
By default prints too much, over terminal height:
- https://github.com/bulletmark/cdhist/issues/7#issuecomment-1319300258

**Comment by bulletmark** [comment]
My question is - why does that matter?

**Comment by glensc** [comment]
what does matter? I don't want this command to mess up my scroll history, therefore limit how much it prints.

**Comment by glensc** [comment]
seems `-m` does what I want. For future readers `cdhist --lines=LINES` would be the same as:

```
cdhist -m $((LINES-2)) -l
```

**Comment by bulletmark** [comment]
So you are only concerned about `-l/--list`? What about `--` which is much more commonly used? Please be clear to explain issues when you raise them.

**Comment by bulletmark** [comment]
No, `-m` is not at all the same. That limits the size of the history kept so immediately you run that you delete all older entries from your stack in `~/.cd_history`. So run it in a tiny terminal window and it is truncated forever more then for all other windows. 

I am trying to get you to explain what you really need. Perhaps you want an option to "limit the output of `-/--list` and `--` to the size of your terminal (with a +/- offset)". I personally don't care if it runs off the top of the window and at least I can scroll back if I really do want to see older entries.

**Comment by glensc** [comment]
But I described what I want: `--list=LINES`, print `LINES` amount of items.

I don't think this tool should be smart and detect terminal size. it's not its purpose. and perhaps tomorrow I want `$LINES-4`, because my prompt is now 3 lines long.

**Comment by bulletmark** [comment]
So (as I asked above) you want to limit the height for the `-l/--list` case but you don't care about limiting the height for the `--` case? Users in general would much more frequently use `--`.

I try another guess - are you wanting a separate option `-n/--num-lines` option to limit to the specified number of lines of output for both those options?

**Comment by glensc** [comment]
I did not understand what you mean by `--`. seems it's an option like `--list`. I don't use it, so I don't care about it (no offense).

`--list=LINES` seems logical use than `--list --num-lines=$LINES`.

but don't know how you could do the same with `--`, `--$LINES` breaks the concept of `--`, it  should denote "stop parsing arguments, pass rest of them to program".

**Comment by glensc** [comment]
btw, how do you know what is commonly used, `-l` or `--`? did you made a survey?

I personally use `cd -l` (alt+p bash bindkey actually), because it came with `cdhist.bash` from bash2-doc:
- https://opensource.apple.com/source/bash/bash-106.201.1/bash-3.2/examples/scripts.v2/cdhist.bash.auto.html

**Comment by bulletmark** [comment]
Surely you have read the main README? It has always explained to users that "all you really need to know" is the `--` command. It combines both `-l` and `-n` options in one go so I don't understand why anybody would not use it. So are you saying that you type `cd -l` and then `cd -n` everytime? So for the reason that it is how the README describes, and it is just way more efficient, I would expect by far most users would use `--`.

I suspect you have used Linux for a while like myself so you should know it is not convention to allow command options to be specified as a boolean switch option, and to also take an argument. In fact, the standard Python argument parser does not allow it. For that reason I am not going to make `-l` do that and have to add the limit as a separate option. You are not going to specify `-l $LINES` by hand every time so I suspect you are adding that with a wrapper alias/function and thus it is irrelevant whether it is a second option anyhow. Can you please be clear in your description of how you want to use this.

I used `--` because I wanted something very easy and quick to type and which invokes cdhist interactively. However, it does work as a user would expect to terminate options, e.g. if you type `cd -- -l`. That would simply `cd` into the directory named `-l`.

**Comment by glensc** [comment]
Added to description:
> The idea is to limit output of cdhist by a NUMBER.

Would that be clear enough?

and yes, I have it as `bindkey` in my `.zshrc` as already [revealed](https://github.com/bulletmark/cdhist/issues/8#issuecomment-1319367113):

```zsh
# cdhist: https://github.com/bulletmark/cdhist
source <(cdhist -i)
# https://jdhao.github.io/2019/06/13/zsh_bind_keys/
bindkey -s "\ep" 'cdhist -m $((LINES-2)) -l\n'
```

also [counter approve](https://github.com/bulletmark/cdhist/issues/8#issuecomment-1319342178) about cdhist trying to figure out terminal height, proof already next day I had to change it:
- https://gitlab.com/glensc/okas/-/commit/d10b12aa0c8f76c9b37aefb4c074202d69bf5361

> you should know it is not convention to allow command options to be specified as a boolean switch option

have to argue about it. optional values are allowed. very common.
for example `git log`:  `--no-decorate, --decorate[=short|full|auto|no]`

and supported by getopt from util-linux. the `::` syntax:

```
TEMP=$(getopt -o 'ab:c::' --long '-long,b-long:,c-long::' -n 'example.bash' -- "$@")
```

- https://github.com/util-linux/util-linux/blob/d27f5fe770025e456fbd33b87a96968f34c455ed/misc-utils/getopt-parse.bash#L23

quick google revealed you could use `nargs` and `str2bool` for Python argparse:

```py
parser.add_argument("--nice", type=str2bool, nargs='?',
                        const=True, default=False,
                        help="Activate nice mode.")
```

```sh
script --nice
script --nice <bool>
```

- https://stackoverflow.com/a/43357954/2314626

**Comment by bulletmark** [comment]
You seem to be missing the point that I have to deal with **both** the `-l/--list` and `--` cases so my proposal is to add a new option `-n/--num-lines` to limit output to the specified number of lines. Would that satisfy your use? I suspect so because you can just set `bindkey -s "\ep" 'cdhist -n $((LINES-2)) -l\n'`

**Comment by glensc** [comment]
Yes. I mumbled that adding `--<number>` seems problematic, and doesn't matter really to me if I have to use `-n <number> -l`. so go ahead with `-n`.

**Comment by bulletmark** [comment]
Added with commit 62598f0f6449d8e2dbcd2bc212722fa0b863a079 and released as version [3.4](https://github.com/bulletmark/cdhist/releases/tag/3.4).
